src/disk_interface.cc

```diff
@@ -116,15 +116,14 @@ bool StatAllFilesInDir(const string& dir, map<string, TimeStamp>* stamps,
     return false;
   }
   do {
-    string lowername = ffd.cFileName;
-    if (lowername == "..") {
+    if (ffd.cFileName[0] == '.' && ffd.cFileName[1] == '.') {
       // Seems to just copy the timestamp for ".." from ".", which is wrong.
       // This is the case at least on NTFS under Windows 7.
       continue;
     }
-    transform(lowername.begin(), lowername.end(), lowername.begin(), ::tolower);
-    stamps->insert(make_pair(lowername,
-                             TimeStampFromFileTime(ffd.ftLastWriteTime)));
+    std::string lowername = ffd.cFileName;
+    std::transform(lowername.begin(), lowername.end(), lowername.begin(), ::tolower);
+    stamps->emplace(std::move(lowername), TimeStampFromFileTime(ffd.ftLastWriteTime));
```

Comment:
FYI, `emplace()` checks for duplicates _after_ creating a temporary `value_type`, so this change does not avoid a string copy after all. A solution is to use the C++11 equivalent of `try_emplace()` (available since C++17), such as:

```
// C++11 equivalent to stamps->try_emplace(std::move(lowername), TimeStamp(..))
stamps->emplace(std::piecewise_construct,
                std::forward_as_tuple(std::move(lowername)),
                std::forward_as_tuple(TimeStampFromFileTime(ffd.ftLastWriteTime)));
```

Another potential optimization on line 186 to avoid a double-lookup + string copy on insertion is:

```
auto ret = cache_.emplace(std::piecewise_construct,
                          std::forward_as_tuple(std::move(dir_lowercase)));
Cache::iterator ci = ret.first;
if (ret.second) {
  if (!Stat(...)) {
    cache_.erase(ci);
    return -1;
  }
}
```

Hope this helps!